core/src/main/java/org/mule/runtime/core/api/util/monitor/ExpiryMonitor.java

```diff
@@ -116,28 +116,32 @@ public void resetExpirable(Expirable expirable) {
   @Override
   public void run() {
     if (!onPollingNodeOnly || muleContext == null || muleContext.isPrimaryPollingInstance()) {
-      for (ExpirableHolder holder : monitors.values()) {
-        if (holder.isExpired()) {
-          removeExpirable(holder.getExpirable());
-          holder.getExpirable().expired();
+      synchronized (monitors) {
+        for (ExpirableHolder holder : monitors.values()) {
+          if (holder.isExpired()) {
+            removeExpirable(holder.getExpirable());
+            holder.getExpirable().expired();
+          }
         }
       }
     }
   }
 
   @Override
   public void dispose() {
-    logger.info("disposing monitor");
+    LOGGER.info("disposing monitor");
     scheduler.stop();
     ExpirableHolder holder;
-    for (Object element : monitors.values()) {
-      holder = (ExpirableHolder) element;
-      removeExpirable(holder.getExpirable());
-      try {
-        holder.getExpirable().expired();
-      } catch (Exception e) {
-        // TODO MULE-863: What should we really do?
-        logger.debug(e.getMessage());
+    synchronized (monitors) {
+      for (Object element : monitors.values()) {
+        holder = (ExpirableHolder) element;
+        removeExpirable(holder.getExpirable());
+        try {
+          holder.getExpirable().expired();
+        } catch (Exception e) {
+          // TODO MULE-863: What should we really do?
```

Comment:
Does this TODO come from mule 1.x? The jira number is prety old. If that is correct, remove it